### PR TITLE
Clone maps

### PR DESCRIPTION
**Pull request by Midnighter**
I went through all cases in the pipeline where I could see a `Map` being modified. As far as I could see, we clone maps where necessary.

I tried to clean up the syntax a bit here and there but didn't do it in all places. So maybe that should be removed again.

I also marked with `FIXME` comments, where channel joins or groupings happen which are hot spots for further improvements in future. I can also take those out again.

### CODE REVIEW

File: subworkflows/local/db_check.nf [hunk 1]
```diff
@@ -28,22 +28,20 @@ workflow DB_CHECK {
...
+        .map { row ->
+            validate_db_rows(row)
+            return [ row.subMap(['tool', 'db_name', 'db_params']), file(row.db_path) ]
         }
 
     ch_dbs_for_untar = parsed_samplesheet
-        .branch {
-            untar: it[1].toString().endsWith(".tar.gz")
+        .branch { db_meta, db ->
+            untar: db.name.endsWith(".tar.gz")
```
[the remark this answers is not in the document]
Comment (Midnighter) [reply subworkflows/local/db_check.nf:38]: `db` is the parameter to the closure which in this case is a file path. The path has the attribute `.name`.

[the remark this answers is not in the document]
Comment (sofstam) [reply subworkflows/local/db_check.nf:38]: Ok, thank you.

File: subworkflows/local/longread_preprocessing.nf [hunk 2]
```diff
@@ -20,33 +20,23 @@ workflow LONGREAD_PREPROCESSING {
         PORECHOP_PORECHOP ( reads )
 
         ch_processed_reads = PORECHOP_PORECHOP.out.reads
-                                        .map {
-                                                meta, reads ->
-                                                def meta_new = meta.clone()
-                                                meta_new['single_end'] = 1
-                                                [ meta_new, reads ]
-                                        }
+            .map { meta, reads -> [ meta + [single_end: 1], reads ] }
```
Comment (jfy133) [inline subworkflows/local/longread_preprocessing.nf:23]: so this isn't additive, `single_end` will overwrite what it is in `meta`?

File: subworkflows/local/longread_preprocessing.nf [hunk 3]
```diff
@@ -20,33 +20,23 @@ workflow LONGREAD_PREPROCESSING {
...
-                                                meta, reads ->
-                                                def meta_new = meta.clone()
-                                                meta_new['single_end'] = 1
-                                                [ meta_new, reads ]
-                                        }
-
-        ch_processed_reads = FILTLONG ( ch_clipped_reads.map{ meta, reads -> [meta, [], reads ]} ).reads
+            .map { meta, reads -> [ meta + [single_end: 1], reads ] }
+
+        ch_processed_reads = FILTLONG ( ch_clipped_reads.map { meta, reads -> [meta, [], reads ] } ).reads
```
Comment (jfy133) [inline subworkflows/local/longread_preprocessing.nf:39]:
```suggestion
        ch_processed_reads = FILTLONG ( ch_clipped_reads.map { meta, reads -> [ meta, [], reads ] } ).reads
```

File: subworkflows/local/input_check.nf [hunk 4]
```diff
@@ -37,49 +37,40 @@ workflow INPUT_CHECK {
...
-}// Function to get list of [ meta, fasta ]
+}
+
+// Function to get list of [ meta, fasta ]
 def create_fasta_channel(LinkedHashMap row) {
-    def meta = [:]
-    meta.id                     = row.sample
-    meta.run_accession          = row.run_accession
-    meta.instrument_platform    = row.instrument_platform
+    def meta = row.subMap(['sample', 'run_accession', 'instrument_platform'])
```
Comment (jfy133) [inline subworkflows/local/input_check.nf:68]: I think this is breaking the tests -> `sample` should be recorded as `id`

**Review by jfy133 — CHANGES_REQUESTED** [review]

**Review by jfy133 — COMMENTED** [review]
After speaking in gather: the FIXMEs aren't necessary

File: subworkflows/local/longread_preprocessing.nf (hunk 2 above)
Comment (jfy133) [reply subworkflows/local/longread_preprocessing.nf:23]: Explained it does in perosn :)

**Review by jfy133 — APPROVED** [review]

**Pull request merged**